addons/web_editor/controllers/main.py

```diff
@@ -224,6 +224,12 @@ def add_url(self, url, res_id=False, res_model='ir.ui.view', **kwargs):
         attachment = self._attachment_create(url=url, res_id=res_id, res_model=res_model)
         return attachment._get_media_info()
 
+    @http.route('/web_editor/attachment/fetch', type='json', auth='user', methods=['POST'], website=True)
+    def fetch_attachments(self, domain=None, fields=None, offset=0, limit=None, order=None, **read_kwargs):
+        if not request.env.user.has_group('base.group_user'):
```

Comment:
`route()` is already in `auth='user'?

Reply:
This is to check if the user is a portal user

Reply:
Yes, but the portal user doesn't have group `base.group_user` so we will just return an empty list.
Previously we had a different code, something like:
```py
Attachment = request.env['ir.attachment']
if request.env.user.has_group('base.group_portal'):
    domain = expression.AND([[['create_uid', '=', request.env.uid]], domain])
    Attachment = Attachment.sudo()
return Attachment.search_read(domain, fields, offset, limit, order, **read_kwargs)
```
This code added a filter to the domain, so portal users were only able to fetch their own attachments. But because all the attachments he creates are not public, this will return nothing.
So we can simply return nothing